src/main/java/com/petfinder/controller/AdvertisementController.java

```diff
@@ -54,55 +69,201 @@ public String newAdvertisement(@RequestParam(required = false) String title,
+        Advertisement advertisement = new Advertisement(title, content, null, pet, location, null, null);
+
+        model.addAttribute("advertisement", advertisement);
+        model.addAttribute("addId", idAdd);
+        model.addAttribute("tags", tagsString);
+        model.addAttribute("files", advertisementService.getAttachmentNameForAdvertisement(idAdd));
+        model.addAttribute("categories", advertisementService.getAllCategories());
+        return "editAdvertisement";
+    }
+
+    @RequestMapping(value = "/getEditAdd")
+    public String getToEditAdvertisement(@RequestParam(required = false) Long id, Model model) {
+        try {
+            Advertisement advertisement = advertisementService.getToEditAdvertisement(id);
+            List<Tag> tagsList = advertisement.getTags();
+            String currentTagsString = "";
+            for (Tag tag : tagsList) {
+                if (tagsList.indexOf(tag) != 0) {
+                    currentTagsString += ",";
+                }
+                currentTagsString += tag.getName();
+            }
+            model.addAttribute("advertisement", advertisement);
+            model.addAttribute("tags", currentTagsString);
+            model.addAttribute("categories", advertisementService.getAllCategories());
+            model.addAttribute("files", advertisementService.getAttachmentNameForAdvertisement(id));
+            model.addAttribute("addId", id);
+        } catch (UserDoesNotHavePermissionToAdvertisemntException e) {
+            LOGGER.log(Level.SEVERE, "UserDoesNotHavePermissionToAdvertisemntException is returned");
+            model.addAttribute("status", e.getMessage());
+        }
+        return "editAdvertisement";
+    }
+
+    @ResponseBody
+    @RequestMapping(value = "/searchResult")
+    public ModelAndView getSearchResults(Model model,
+            @RequestParam(required = false) String adInfo,
+            @RequestParam(required = false) String petInfo,
+            @RequestParam(required = false) String locationInfo,
+            @RequestParam(required = false) String tagInfo,
+            @RequestParam(required = false) int page
+    ) {
         model.addAttribute("advertisements",
-                advertisementService.getSearchedAdvertisements(page-1, 20, adInfo, petInfo, locationInfo, tagInfo)
+                advertisementService.getSearchedAdvertisements(page - 1, 20, adInfo, petInfo, locationInfo, tagInfo)
         );
         this.preparePagination(model, page);
         model.addAttribute("adInfo", adInfo);
         model.addAttribute("petInfo", petInfo);
         model.addAttribute("locationInfo", locationInfo);
         model.addAttribute("tagInfo", tagInfo);
-        return new ModelAndView( "searchResults" );
+        return new ModelAndView("searchResults");
     }
-	
-	private void preparePagination(Model model, int page)
-	{
+
+    private void preparePagination(Model model, int page) {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [The method preparePagination() has an NPath complexity of 400](https://www.codacy.com/app/matikz1110/PetFinder/file/2653733612/issues/source?bid=3239464&fileBranchId=3326233#l266)